core/lib/server/pub.ts

```diff
@@ -587,8 +597,37 @@ export const createPubRecursiveNew = async <Body extends CreatePubRequestBodyWit
 	return result;
 };
 
-export const deletePub = (pubId: PubsId, trx = db) =>
-	autoRevalidate(trx.deleteFrom("pubs").where("id", "=", pubId));
+export const deletePub = async ({
+	pubId,
+	lastModifiedBy,
+	trx = db,
+}: {
+	pubId: PubsId;
+	lastModifiedBy: LastModifiedBy;
+	trx?: typeof db;
+}) => {
+	// first get the values before they are deleted
+	const pubValues = await trx
+		.selectFrom("pub_values")
+		.where("pubId", "=", pubId)
+		.selectAll()
+		.execute();
+
+	const deleteResult = await autoRevalidate(
+		trx.deleteFrom("pubs").where("id", "=", pubId)
+	).executeTakeFirstOrThrow();
+
+	// this might not be necessary if we rarely delete pubs and
+	// give users ample warning that deletion is irreversible
+	// in that case we should probably also delete the pub values
+	await addDeletePubValueHistoryEntries({
+		lastModifiedBy,
+		pubValues,
+		trx,
+	});
```

Comment:
i don't like doing this, see pr description for alternatives.

Reply:
yes, you're right. what i mean to say is that we should also delete the pub_values_history table!